src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -2337,5 +2341,44 @@ private IInstanceReferenceOperation CreateCollectionValuePlaceholderOperation(Bo
             bool isImplicit = placeholder.WasCompilerGenerated;
             return new InstanceReferenceOperation(referenceKind, _semanticModel, syntax, type, isImplicit);
         }
+
+        private ImmutableArray<IArgumentOperation> CreateDisposeArguments(MethodSymbol? disposeMethod, TypeSymbol? receiverType, SyntaxNode receiverSyntax, Binder binder)
+        {
+            if (disposeMethod is null || receiverType is null || disposeMethod.ParameterCount == 0)
+            {
+                return ImmutableArray<IArgumentOperation>.Empty;
+            }
+
+            var receiver = new BoundDisposableValuePlaceholder(receiverSyntax, receiverType);
+
+            var argumentsBuilder = ArrayBuilder<BoundExpression>.GetInstance(disposeMethod.ParameterCount);
+            ImmutableArray<int> argsToParams = default;
+            var expanded = disposeMethod.HasParamsParameter();
```

Comment:
>HasParamsParameter [](start = 41, length = 18)

This looks wrong. This helper doesn't perform the check consistent with OverloadResolution.IsValidParams. It also looks like all other usages have the same problem, possibly with exception of ExplicitInterfaceHelpers because they only affect diagnostics in a very restricted scenario. I think we should open an issue to clean all this up, but here we should do the right thing.

Reply:
Ideally we should capture this value during initial binding. I assume we performed overload resolution over available candidates, that should have given us the information about the best candidate and whether it should be used in an expanded form. We should not be trying to re-infer this aspect again.

---
In reply to: [539480603](https://github.com/dotnet/roslyn/pull/49842#discussion_r539480603) [](ancestors = 539480603)